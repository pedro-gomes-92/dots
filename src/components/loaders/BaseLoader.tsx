import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseProps, Base } from '../Base';
import { IconProps } from '../miscellaneous/Icon';

export interface BaseLoaderProps extends BaseProps {}

export interface BaseLoaderTypeProps extends BaseLoaderProps {
  sizeIcon?: IconProps['size'];
  icon?: IconProps['name'];
}

export const BaseLoader: FunctionComponent<BaseLoaderProps> = ({
  className,
  children,
  ...rest
}: BaseLoaderProps): JSX.Element => {
  return (
    <Base className={classnames('loaders', className)} {...rest}>
      {children}
    </Base>
  );
};
