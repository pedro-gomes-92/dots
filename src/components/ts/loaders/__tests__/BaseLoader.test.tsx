import React from 'react';
import { render } from '@testing-library/react';

import { BaseLoader } from '../BaseLoader';

describe('Given <BaseLoader />', (): void => {
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<BaseLoader />);
      expect(container).toMatchSnapshot();
    });
  });
});
