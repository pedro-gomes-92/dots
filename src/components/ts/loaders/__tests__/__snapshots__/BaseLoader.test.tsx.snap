// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <BaseLoader /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="loaders"
  />
</div>
`;
